spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/KerasLayer.scala

```diff
@@ -167,6 +167,75 @@ abstract class KerasLayer[A <: Activity: ClassTag, B <: Activity: ClassTag, T: C
 
   inputShapeValue = batchInputShape
 
+  override def getEndNodes(startNodes: Array[ModuleNode[T]]): Array[ModuleNode[T]] = {
+    if (this.isKerasGraph()) {
+      this.toGraph().getEndNodes(startNodes)
+    } else if (labor.isKerasStyle() && labor.getName().equals(this.getName())) {
+      Array(this.toGraphInputs(startNodes))
```

Comment:
if `inputs(startNodes)` is not suitable for keras layer, you can define a new one here not in `AbstractModule`

Reply:
The purpose of toGraphInputs() is to avoid exclusive check (the one which bans connections between blas nodes and keras nodes). I wonder if moving it to KerasLayer would only handle blas-to-keras connection, but not keras-to-blas connection?

Reply:
I think you can also call `toGraphInputs` instead of  `inputs`  for keras-to-blas case. And all process should be only limited to `KerasLayer`。